pkg/wasm/convert.go

```diff
@@ -144,6 +145,20 @@ func tryUnmarshal(resource *anypb.Any) (*core.TypedExtensionConfig, *wasm.Wasm,
 			wasmLog.Debugf("typed extension config %+v does not contain wasm http filter", typedStruct)
 			return nil, nil, nil
 		}
+	case ec.GetTypedConfig().TypeUrl == xds.ExtensionWithMatcher:
```

Comment:
I'm confused how this works end-to-end? We need to preserve the matcher and rewrite Wasm config, so I expect the matcher to be somehow re-created?